core/ops.rs

```diff
@@ -20,19 +22,84 @@ use std::ops::DerefMut;
 use std::pin::Pin;
 use std::rc::Rc;
 
-pub type OpAsyncFuture = Pin<Box<dyn Future<Output = Box<[u8]>>>>;
-pub type OpFn = dyn Fn(Rc<RefCell<OpState>>, BufVec) -> Op + 'static;
+pub use erased_serde::Serialize as Serializable;
+pub type RcOpState = Rc<RefCell<OpState>>;
```

Comment:
Please remove, this is additional congnitive load compared to using `Rc<RefCell<OpState>>` directly

Reply:
I think `RcOpState` is substantially more legible than `Rc<RefCell<OpState>>`. The fully expanded func-signature would be verbose here:
```
F: Fn(RcOpState, OpPayload, OpBuf) -> Op + 'static,
```

I think as a general rule type-aliases should be discouraged, but when you've got one place that defines a type and then N other users (were N is large ~200 here) that don't care about the type specifics, the type is more an interface than a concrete type in their eyes, since it's mainly "however state is passed to OpFn".

`OpBuf` follows a similar rationale, 199/220 or 90% of our ops don't use the data buffer arg, they don't care about the type specifics and simply want to follow / implement an interface.

The `OpBuf` "issue" could be worked around by adding buffer-less json ops (essentially what `tsc` does), but that causes its own fragmentation and issues IMO.

Here's dump of of deno's op signatures: https://gist.github.com/AaronO/2c03a99912b786c7a3f27f2b9f9bf9a5